src/main/webapp/scripts/matrixController.js

```diff
@@ -1,32 +1,179 @@
 var rowOpening = "<tr>";
 var rowClosure = "</tr>";
 var cellZoom = "style = \"width: 3em; height: 3em\"";
-var alive = "<td class = \"matrixCell\"><img src = \"../images/alive normal cell.png\" " + cellZoom+"class = \"table-image\"></td>";
-var death = "<td class = \"matrixCell\"><img src = \"../images/dead cell.png\" "+cellZoom+"class = \"table-image\"></td>";
+var alive = "<td class = \"matrixCell\"><img src = \"../images/alive normal cell.png\" " + cellZoom+"class = \"table-image\" ";
+var death = "<td class = \"matrixCell\"><img src = \"../images/dead cell.png\" "+cellZoom+"class = \"table-image\" ";
+var cellClosure = "></td>";
 var rows = 0;
 var columns = 0;
-var matrix = "";
+var selectedCellType = "deadCell"
+var matrix = [[0,0,0],[0,0,0],[0,0,0]];
+var playSpeed = 1000;
+var cellWithMask = false;
+var cellWithVaccine = false;
+var playing = false;
+var timeController = null;
+var cellDictionary ={
+    0: "../images/dead cell.png",
+    1: "../images/alive normal cell.png",
+    2: "../images/alive mask.png",
+    3: "../images/alive vaccine.png",
+    4: "../images/alive mask vaccine.png",
+    5: "../images/Infected Cell.png",
+    6: "../images/infected mask.png",
+    7: "../images/infected vaccine.png",
+    8: "../images/infected mask vaccine.png"
+};
 
+function generateRandomMatrix(){
+    for(let row = 0; row < rows; row++){
+        for(let column = 0 ; column < columns ; column++ ){
+            document.getElementById(row.toString() + column.toString()).src = cellDictionary[ Math.floor( Math.random() * 8 ) ];
+        }
+    }
+    gameLogic();
+}
 
-function updateMatrix(){
+function gameLogic(){
+    if(playing){
+        timeController = setTimeout(() => {
+                                                if(playing){
+                                                    generateRandomMatrix();
+                                                }
+                                            }, playSpeed);
+    }
+}
+
+function startTheGame(){
+    if(!playing){
+        playing = true;
+        gameLogic();
+    }
+}
+
+function stopTheGame(){
+    playing = false;
+    clearInterval(timeController);
+
+}
+
+function updateCellModifier(target){
+    if(target == "mask")
+        cellWithMask = !cellWithMask;
+    else
+        cellWithVaccine = !cellWithVaccine;
+    console.log(cellWithMask);
+    console.log(cellWithVaccine);
+}
 
+function updateMatrix(){
+    matrix = []
```

Comment:
Fundamentally, it seems like we're representing the state of the matrix twice, once in the matrix variable, and once in the HTML table, and the translation from one to the other is pretty manual. I'd be surprised if there wasn't a framework/library out there that would contain a lot of the complexity for you.